controllers/replica_calculator.go

```diff
@@ -38,6 +38,12 @@ type ReplicaCalculation struct {
 	utilization   int64
 	timestamp     time.Time
 	readyReplicas int32
+	pos           MetricPosition
+}
+
+type MetricPosition struct {
```

Comment:
I think this struct can be private

```suggestion
type metricPosition struct {
```